proxy_agent/src/common/windows.rs

```diff
@@ -19,13 +19,16 @@ pub fn get_processor_count() -> usize {
     data.dwNumberOfProcessors as usize
 }
 
-pub fn get_memory_in_mb() -> u64 {
+pub fn get_memory_in_mb() -> Result<u64, String> {
     let mut data = MaybeUninit::<MEMORYSTATUSEX>::uninit();
     let data = data.as_mut_ptr();
     unsafe {
         (*data).dwLength = std::mem::size_of::<MEMORYSTATUSEX>() as u32;
-        GlobalMemoryStatusEx(data);
-        (*data).ullTotalPhys / 1024 / 1024
+        if GlobalMemoryStatusEx(data) == 0 {
+            return Err("GlobalMemoryStatusEx failed".to_string());
```

Comment:
https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-globalmemorystatusex
If the function fails, the return value is zero. To get extended error information, call [GetLastError](https://learn.microsoft.com/en-us/windows/desktop/api/errhandlingapi/nf-errhandlingapi-getlasterror).

Please log this Error::last_os_error() too.

Reply:
adding logger class to the file and log both errors